src/components/Payment/PaymentForm.tsx

```diff
@@ -158,10 +158,29 @@ const PaymentForm = React.memo(function PaymentForm(props: PaymentFormProps) {
     props.onSubmit(formValues, spendableBalance, matchingWellknownAccount)
   }
 
+  const handleQRScan = React.useCallback((scanResult: string) => {
```

Comment:
We can write this function in a more elegant way (sorry, but the one-click suggestions in comments don't seem to work if you want to change more than one line…)

```ts
  const handleQRScan = React.useCallback((scanResult: string) => {
    const [destination, query] = scanResult.split("?")
    setFormValue("destination", destination)

    if (!query) {
      return
    }

    const searchParams = new URLSearchParams(query)
    const memoValue = searchParams.get("dt")

    // ...
}, [])
```

Reply:
Looks fine otherwise – still need to test it.

Reply:
I added your suggested change.